main.go

```diff
@@ -413,6 +450,51 @@ func convert(req *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, e
 	return res, nil
 }
 
+func addExtensions(msg *descriptor.DescriptorProto, extensions []*descriptor.FieldDescriptorProto, pkg string) {
+	for _, ext := range extensions {
+		extendeeNodes := strings.Split(ext.GetExtendee(), ".")
+		pkgNodes := strings.Split(pkg, ".")
+
+		// We already know this extendee cannot be in this package.
+		if len(pkgNodes) > len(extendeeNodes) {
+			continue
+		}
+
+		// skipping leading "."
+		nodeIdx := 1
+		for _, node := range pkgNodes {
+			if node != extendeeNodes[nodeIdx] {
```

Comment:
that will fail if there is no dot in `ext.GetExtendee()`. I would recommend to start at `nodeIdx := 0` and do a len check on extendeeNodes above:

```go
if len(extendeeNodes) < 1 {
    continue
}
extendeeNodes = extendeeNodes[1:]
```

Reply:
Yes, I did that initially, but then I learned that it always has a dot.  But sure, I can revert back

Reply:
As in, let's not trust whatever source it's coming from. If one day the `.` isn't there, we'll get panic all over the place haha